### PR TITLE
Store migrations in a migrations db

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [x] Build/deploy pipeline (DevOps)
- [x] Other

## Objective
Improves migration storage for dev and pipeline work.

Added the -p flag to run_migrations for pipeline to run from environment variables

Created meta-migrations script to pre-populate migrations_vault_dev tables with run migrations for those already using the last_migrations file

I will update https://bitwarden.atlassian.net/wiki/spaces/DeveloperDocs/pages/164004068/Server+Setup+Guide#Create-database when merged in.

## Before you submit
- [x] I have checked for formatting errors (`dotnet tool run dotnet-format --check`) (required)
- [ ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [x] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Review by Hinton — APPROVED** [review]
We're essentially re-creating EF migration logic 😭. How is the performance, I guess it's negligible to run a couple of queries?

File: dev/helpers/mssql/migrate_migrations.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+sleep 0.1;
+
+MIGRATE_DIRECTORY="/mnt/migrator/DbScripts"
+LAST_MIGRATION_FILE="/mnt/data/last_migration"
+SERVER='mssql'
+DATABASE="vault_dev"
+USER="SA"
+PASSWD=$MSSQL_PASSWORD
+
+while getopts "rs" arg; do
```
Comment (vgrassia) [inline dev/helpers/mssql/migrate_migrations.sh:15]:
```suggestion
while getopts "s" arg; do
```

File: dev/helpers/mssql/run_migrations.sh [hunk 2]
```diff
@@ -6,53 +6,64 @@
 sleep 0.1;
 
 MIGRATE_DIRECTORY="/mnt/migrator/DbScripts"
-LAST_MIGRATION_FILE="/mnt/data/last_migration"
 SERVER='mssql'
 DATABASE="vault_dev"
 USER="SA"
 PASSWD=$MSSQL_PASSWORD
 
 while getopts "rs" arg; do
```
Comment (vgrassia) [inline dev/helpers/mssql/run_migrations.sh:14]:
```suggestion
while getopts "sp" arg; do
```

File: dev/helpers/mssql/run_migrations.sh [hunk 3]
```diff
@@ -61,18 +72,11 @@ migrate () {
 }
```
Comment (vgrassia) [inline dev/helpers/mssql/run_migrations.sh:72]: Remove `echo $file > $LAST_MIGRATION_FILE` from the `migrate ()` function above.

**Review by vgrassia — CHANGES_REQUESTED** [review]

**Comment by MGibson1** [comment]
performance is very bad, but yes, this is an unfortunate place we find ourselves in, recreating this for a different db orm

File: dev/helpers/mssql/migrate_migrations.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+else
+  LAST_MIGRATION=$(cat $LAST_MIGRATION_FILE)
+  rm $LAST_MIGRATION_FILE
+fi
+
+[ -z "$LAST_MIGRATION" ]
+PERFORM_MIGRATION=$?
+
+# Create database if it does not already exist
+QUERY="IF NOT EXISTS (SELECT * FROM sys.databases WHERE name = 'migrations_$DATABASE')
```
Comment (justindbaur) [inline dev/helpers/mssql/migrate_migrations.sh:37]: Why a separate database and not just a special table in the same database like EF does it?

Comment (MGibson1) [reply dev/helpers/mssql/migrate_migrations.sh:37]: 🤷 not a super good reason I guess, but I was thinking we use a new db just to not limit the names available for tables for product.

**Pull request merged**